vkcs/networking/subnet.go

```diff
@@ -21,15 +22,47 @@ type subnetExtended struct {
 
 // networkingSubnetStateRefreshFunc returns a standard retry.StateRefreshFunc to wait for subnet status.
 func networkingSubnetStateRefreshFunc(client *gophercloud.ServiceClient, subnetID string) retry.StateRefreshFunc {
-	return func() (interface{}, string, error) {
-		subnet, err := isubnets.Get(client, subnetID).Extract()
+	return func() (any, string, error) {
+		var subnet subnetExtended
+		err := isubnets.ExtractSubnetInto(isubnets.Get(client, subnetID), &subnet)
 		if err != nil {
 			if errutil.IsNotFound(err) {
 				return subnet, "DELETED", nil
 			}
 
 			return nil, "", err
 		}
+		if !subnet.EnableDHCP || subnet.SDN != networking.NeutronSDN {
+			return subnet, "ACTIVE", nil
+		}
+
+		// check dhcp ports
+		var listOpts ports.ListOptsBuilder = ports.ListOpts{
+			ProjectID:   subnet.ProjectID,
+			NetworkID:   subnet.NetworkID,
+			DeviceOwner: "network:dhcp",
+			Status:      "ACTIVE",
+			FixedIPs: []ports.FixedIPOpts{
+				{
+					SubnetID: subnetID,
+				},
+			},
+		}
+
+		allPages, err := ports.List(client, listOpts).AllPages()
+		if err != nil {
+			return nil, "", fmt.Errorf("error listing ports of subnet: %s", err)
+		}
+
+		var allPorts []ports.Port
+		err = ports.ExtractPortsInto(allPages, &allPorts)
+		if err != nil {
+			return nil, "", fmt.Errorf("error reading VKCS Networking API response: %s", err)
+		}
+
+		if len(allPorts) < 2 {
+			return subnet, "WAITING_DHCP_PORTS", nil
```

Comment:
You should add this status to list of expected statuses or you will get an error, don't you? And do we have such test for creating a neutron subnet with enabled dhcp?

Reply:
we do not specify pending statuses for subnet, because of this it's working. Tests are passed + i test this locally

Reply:
dhcp is enable by default, all our tests will test it